lib/plug/static.ex

```diff
@@ -112,32 +146,35 @@ defmodule Plug.Static do
     end
   end
 
+  @spec gzip?(Plug.Conn.t) :: boolean
   defp gzip?(conn) do
-    fun = &(:binary.match(&1, ["gzip", "*"]) != :nomatch)
+    gzip_header? = &String.contains?(&1, ["gzip", "*"])
     Enum.any? get_req_header(conn, "accept-encoding"), fn accept ->
-      Enum.any?(Plug.Conn.Utils.list(accept), fun)
+      accept |> Plug.Conn.Utils.list() |> Enum.any?(gzip_header?)
     end
   end
 
+  @spec path(Path.t | atom, [String.t]) :: Path.t
+  defp path(from, segments)
+
   defp path(from, segments) when is_atom(from),
     do: Path.join([Application.app_dir(from), "priv/static" | segments])
 
   defp path(from, segments),
     do: Path.join([from | segments])
 
+  @spec subset([String.t], [String.t]) :: [String.t]
```

Comment:
We don't `@spec` private functions anywhere else in Plug, so I wouldn't spec them here.